records/validation_test.go

```diff
@@ -58,15 +44,22 @@ func TestValidateResolvers(t *testing.T) {
 		{[]string{"2001:0db8:3c4d:0015:0000:0000:1a2f:1a2b"}, true},
 		{[]string{"2001:db8:3c4d:15::1a2f:1a2b"}, true},
 		{[]string{"2001:0db8:3c4d:0015:0000:0000:1a2f:1a2b", "2001:db8:3c4d:15::1a2f:1a2b"}, false},
+	} {
+		validate(t, i+1, tc, validateResolvers)
 	}
-	for i, tc := range table {
-		err := validateResolvers(tc.rs)
-		if (err == nil && tc.valid) || (err != nil && !tc.valid) {
-			continue
-		} else if tc.valid {
-			t.Fatalf("test %d failed, unexpected error validating resolvers %v: %v", i+1, tc.rs, err)
-		} else {
-			t.Fatalf("test %d failed, expected validation error for resolvers(%d) %v", i+1, len(tc.rs), tc.rs)
-		}
+}
+
+type validationTest struct {
+	in    []string
+	valid bool
+}
+
+func validate(t *testing.T, i int, tc validationTest, f func([]string) error) {
+	if err := f(tc.in); (err == nil && tc.valid) || (err != nil && !tc.valid) {
+		return // valid
+	} else if tc.valid {
```

Comment:
This is a pure refactoring with the intention to clear warnings in the CircleCI build. I didn't intend to clean things out unless it cleared a warning.